### PR TITLE
move AWS ECR gc jobs to circleci

### PR DESCRIPTION
**Pull request by mingbowan**
all jobs are currently running with "--dry-run", so you can verify if the jobs are doing the right thing.  i'll remove the flag and make it runs every hour same as on Jenkins once this PR is approved.

### CODE REVIEW

**Comment by ezyang** [comment]
Now that the tags to keep are stored directly in this repository, you'll need to adjust this linter:

```
      - name: Ensure Docker version is correctly deployed
        run: .circleci/validate-docker-version.py
```

This linter is still consulting ossci-job-dsl for the most recent versions. You should point it at the new tag information stored locally in this repository.

File: .circleci/config.yml [hunk 1]
```diff
@@ -1626,6 +1626,51 @@ jobs:
...
+            no_output_timeout: "1h"
+            command: |
+              cd .circleci/ecr_gc_docker
+              docker build . -t 308535385114.dkr.ecr.us-east-1.amazonaws.com/gc/ecr
+              set +x
+              export AWS_ACCESS_KEY_ID=${CIRCLECI_AWS_ACCESS_KEY_FOR_DOCKER_BUILDER_V1}
+              export AWS_SECRET_ACCESS_KEY=${CIRCLECI_AWS_SECRET_KEY_FOR_DOCKER_BUILDER_V1}
+              eval $(aws ecr get-login --no-include-email)
+              set -x
+              docker push 308535385114.dkr.ecr.us-east-1.amazonaws.com/gc/ecr
```
Comment (ezyang) [inline .circleci/config.yml]: I wonder if we shouldn't just roll this into the docker build job as a "plain old" image, like any other, so we don't have to special case the build job here. Doesn't have to be done in this PR.

**Review by ezyang — APPROVED** [review]
Looks legit, thank you. Please fix (or disable) the lint job before landing

**Comment by kostmo** [comment]
## CircleCI build failures summary
As of commit 5bf6e31:
* **1/1** broken upstream at merge base bcb0bb7 (see [grid view](https://dr.pytorch.org/master-timeline.html?count=1&sha1=bcb0bb7e0e03b386ad837015faba6b4b16e3bfb9&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35))
    * You may want to rebase on the `viable/strict` branch (see its [recency history](https://dr.pytorch.org/master-viable-commits.html)):
        * If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
          ```
          git fetch viable/strict
          git rebase --onto viable/strict $(git merge-base origin/master HEAD)
          ```
        * If your commit is older than `viable/strict`:
          ```
          git fetch viable/strict
          git rebase viable/strict
          ```
* **0/1** failures introduced in this PR



## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=5bf6e310aac811f7aaa3637d93f32c5f1a7441be).

### 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably caused by upstream breakages:
* [pytorch_xla_linux_xenial_py3_6_clang7_build](https://dr.pytorch.org/build-details.html?build_id=73458684)



---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

**Comment by mingbowan** [comment]
the jobs are good: https://circleci.com/gh/pytorch/pytorch/3932318

**Pull request closed without merging**